### ISSUE TITLE
Wrong x-axis limit in the plotTrainingProgress.py script

### ISSUE THREAD

**Issue by gsanroma** [Kamnitsask/deepmedic#159]
The x-axis limit is wrong, only in the lower-row (training).
This is because not initializing `maxNumOfEpsDurationOfExps` variable before plotting the lower-row.
Attaching an example of output plot.
[trainingProgress.pdf](https://github.com/Kamnitsask/deepmedic/files/3042262/trainingProgress.pdf)

**Comment by Kamnitsask** [comment]
Hello, and thank you for the help.

Actually I am not sure this is a validation/training issue. It may be something else that confused the scripts:

I noticed in your attached pdf above that the top-right validation plot (validation Dice computed over the validation volumes after they are fully segmented) is also getting the wrong value (~180 epochs instead of 40, just like the training plots).

There is something weird:
- From what I understand, you have run an experiment for 40 epochs.
- The top line in the plot says you fully segment the validation cases every 5 epochs. This value is parsed from the log of the experiment/training-session, and the value is set in the trainConfig.cfg that was used for the experiment. (If multiple entries in the log describe this variable, then the first one is parsed. There can be more than one entries eg if you restarted the experiment but didnt delete the logfile).
- The top right plot (DICE from full segmentation of validation cases) has roughly 30-40 measurements. And wrong X-axis.

My thoughts:
> For a 40 epochs training, if full-segmentation of val cases is done every 5 epochs, then top-right should have 8 points. It has much more than this.
> What I believe has happened and has confused the script is that multiple training-sessions were done, with a different value for the parameter 'numberOfEpochsBetweenFullInferenceOnValImages' in the trainConfig. Something like:
- You started one experiment with this variable = 5 (now it's the first entry in the logs, and parsed for plotting).
- Stop this experiment (even very early, even without any epoch completing, but right after the above line was logged).
- Changed the above parameter in the trainConfig to something like = 1.
- Restart the experiment and run till the end (which created the 30+ points in the top right plot).

Does any of the above make sense and ring a bell?
If you send me the log file of your experiment with an email, it would help me massively to understand what happened.

konstantinos.kamnitsas12@imperial.ac.uk

I hope the above makes sense.

**Comment by gsanroma** [comment]
Hi Konstantinos, thank you for the reply and sorry for the delay.
Indeed what you describe is what happened. One training session was started with config variable epochsbetweenfullinference = 5 and immediately stopped. Then I re-set the value to =1 and re-start the training, **without deleting the log file** (for some reason, I assumed that the log file was overwritten when a new run with the same name but only realized later that this is not the case).
I see that `plotTrainingProgress.py` updates the value maximum number of epochs between the validation and training plots (variable `maxNumOfEpsDurationOfExps`). Then, the x-axis gets screwed in the training plots (after the update). The solution proposed in the pull-request is to move the initialization of `maxNumOfEpsDurationOfExps = 0` from the outside to the beginning of the 1st loop so that it gets initialized again after the _problematic_ update between validation and train plots.
Nevertheless, as you correctly noticed, this all stems from a log file containing a sequence of two runs with different value for variable epochsbetweeninference in the config file. Would it make sense to overwrite by default the log files of sessions with the same name ? Or otherwise, is this behavior helpful in other situations (eg, re-starting training session, fine-tunning) ?
